### PR TITLE
Find MIIS

### PR DESCRIPTION
**Pull request by marouenbg**
*Please include a short description of enhancement here*
findMIIS and testfindMIIS finds the Minimal Infeasible Irreducible Submodel in a given infeasible model (solution status =0). Fast and effective tool for model debugging. Only with IBM Cplex for now.
It gives back the reactions and metabolites causing the infeasiblities. The user can then check the original data and correct the reactions accordingly.

**I hereby confirm that I have:**

- [x] Tested my code on my own machine
- [x] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [x] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [] with [X] to check the box)*

### CODE REVIEW

File: test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+%     - Marouen BEN GUEBILA 02/12/2017
+
+global CBTDIR
+global ILOG_CPLEX_PATH
+
+addpath(genpath(ILOG_CPLEX_PATH));
+
+%Test presence of required toolboxes.
+v = ver;
+bioPres = any(strcmp('Bioinformatics Toolbox', {v.Name})) && license('test','bioinformatics_toolbox');
```
Comment (tpfau) [inline test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: Are these toolboxes really required for the test? 
From the code it looks like it should only depend on the CPLEX solver.
You could actually write an assert(changeCobraSolver('ibm_cplex') == 1), but I'm not sure, whether the solver is installed on all CI systems.
If not, the current test will pass without it actually having tested anything.
@laurentheirendt ?

Comment (laurentheirendt) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: Why not do:
```Matlab
solverOK = changeCobraSolver('ibm_cplex');
if solverOK
% ...
end
```

Comment (tpfau) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: Ok, so ibm_cplex is installed on all CI systems? 
I just was not sure about it.

File: src/analysis/findMIIS/findMIIS.m [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+% Output:
+%        MIIS.rxns:     Reactions of MIIS
+%        MIIS.mets:     Mets of MIIS
+%        MIIS.rxnsStat: Status of infeasiblity of reactions
+%        MIIS.metsStat: Status of infeasiblity of metabolites
+%        please refer to this link for status meaning
+% https://www.ibm.com/support/knowledgecenter/de/SSSA5P_12.7.0/ilog.odms.cplex.help/refcallablelibrary/macros/Solution_status_codes.html
+% .. Author: - Marouen Ben Guebila 24/07/2017
+
+if ~changeCobraSolver('ibm_cplex')
```
Comment (laurentheirendt) [inline src/analysis/findMIIS/findMIIS.m]: This should read (output suppressed, silent check):
```Matlab
if ~changeCobraSolver('ibm_cplex', 'LP', 0)
    error('This function requires IBM ILOG CPLEX');
end
```

File: test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+%     an infeasible model and returns to  Minimal Irreducible Infeasible
+%     Submodel
+%
+% Author:
+%     - Marouen BEN GUEBILA 02/12/2017
+
+global CBTDIR
+global ILOG_CPLEX_PATH
+
+addpath(genpath(ILOG_CPLEX_PATH));
```
Comment (laurentheirendt) [inline test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: This is not required, and shouldn't be done. The path is added and set in `changeCobraSolver`.

File: test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m (hunk 1 above)
Comment (laurentheirendt) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: `changeCobraSolver` solver takes care of that.

Comment (laurentheirendt) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: If these toolboxes are required, then this check should be in `findMIIS`, not in the test only.

File: src/analysis/findMIIS/findMIIS.m [hunk 4]
```diff
@@ -0,0 +1,118 @@
+function MIIS = findMIIS(LPProblem,printLevel)
+% Finds the Minimal Irreducible Infeasible Subset (MIIS) in an infeasible
```
Comment (laurentheirendt) [inline src/analysis/findMIIS/findMIIS.m]: Documentation needs formatting.

File: test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m (hunk 1 above)
Comment (marouenbg) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: @tpfau cplex is not installed on windows and mac machines, thus the `if solverOK`

Comment (tpfau) [reply test/verifiedTests/analysis/testfindMIIS/testfindMIIS.m]: ok. 
I just wasn'T sure, and if it ends up not being tested at all that would be inconvenient :)

**Pull request merged**